---
layout: post
title: "神奇'跳过'或’捕获‘错误"
description: "python遇到BUG也勇往直前"
date: 2019-02-06
tags: [learn,python]
comments: true
share: true
---

```python
...

try:
    name = json_data['data'][x]['name']
    player = json_data['data'][x]['player']
    table.add_row([name,player])
except:
    print('Stoping...')
    break
    
...

```

**以上是电影采集一个递归取值的代码，由于API的原因，导致页数和匹配数量有问题，在匹配过程中容易发生列表越界，我百度了一下：python如何跳过列表越界，于是就看到了try和except代码，这个标志在别人的代码里我遇到过，只是不知道具体是做什么用的，今天仔细研究了一下，发现这个代码太神奇了**


## 捕获异常

**try内的代码就是将要‘尝试’执行的代码，except就是当遇到相应的错误时执行的代码，例如except TypeError就是数据类型错误。**

## Such as

![](https://ss2.baidu.com/6ONYsjip0QIZ8tyhnq/it/u=3798518219,225259021&fm=173&app=25&f=JPEG?w=530&h=430&s=A0C2B74BDAA4936C1CF0F40F0000E0C3)

**运行结果：**

![](https://ss1.baidu.com/6ONXsjip0QIZ8tyhnq/it/u=3397314481,530076019&fm=173&app=25&f=JPEG?w=550&h=207&s=E8C2A3410FA089705C55A4030000E0C3)


**expect后面什么也不填，就是捕获所有错误**
